browser/importer/brave_profile_writer.cc

```diff
@@ -63,3 +65,42 @@ void BraveProfileWriter::UpdateStats(const BraveStats& stats) {
                      https_upgrades + stats.httpsEverywhere_count);
   }
 }
+
+void BraveProfileWriter::UpdateReferral(const BraveReferral& referral) {
+  PrefService* local_state = g_browser_process->local_state();
+  if (!local_state) {
+    LOG(ERROR) << "Unable to get local_state! (needed to set referral info)";
+  }
+
+  if (!referral.week_of_installation.empty()) {
+    LOG(INFO) << "Setting kWeekOfInstallation to "
+      << "\"" << referral.week_of_installation << "\"";
+    local_state->SetString(kWeekOfInstallation, referral.week_of_installation);
+  }
+
+  if (!referral.promo_code.empty() &&
+    referral.promo_code.compare("none") != 0) {
+    LOG(INFO) << "Setting kReferralPromoCode to "
+      << "\"" << referral.promo_code << "\"";
+    local_state->SetString(kReferralPromoCode, referral.promo_code);
```

Comment:
What if the imported data is invalid? Does it makes sense to import as much as possible, or would it be better/safer to cancel the referral import unless the data can be completely validated?

Generally, I think it's best to validate the imported data in `BraveImporter`, and only call `BraveProfileWriter` with validated data.